libndt.hpp

```diff
@@ -1960,8 +2015,35 @@ bool Client::ndt7_upload() noexcept {
     constexpr auto measurement_interval = 0.25;
     std::chrono::duration<double> interval = now - latest;
     if (interval.count() > measurement_interval) {
+      nlohmann::json measurement;
+      measurement["AppInfo"] = nlohmann::json();
+      measurement["AppInfo"]["ElapsedTime"] = elapsed.count();
+      measurement["AppInfo"]["NumBytes"] = total;
+#ifdef __linux__
+      // Read tcp_info data for the socket and print it as JSON.
+      struct tcp_info tcpinfo {};
+      socklen_t tcpinfolen = sizeof(tcpinfo);
+      if (sys_getsockopt(sock_, IPPROTO_TCP, TCP_INFO, (void *)&tcpinfo,
+                         &tcpinfolen) == 0) {
+        measurement["TCPInfo"] = nlohmann::json();
+#define XX(lower_, upper_) measurement["TCPInfo"][#upper_] = (uint64_t)tcpinfo.lower_;
```

Comment:
I think we `int64_t` in the spec, so I'd suggest to use `int64_t` here? (Or is there a reason to use `uint64_t`?)

Reply:
Discussed OOB - some of the values are provided by the kernel as `uint64_t`.